symbulate/plot.py

```diff
@@ -40,4 +41,117 @@ def plot(*args, **kwargs):
     except:
         plt.plot(*args, **kwargs)
         
+def is_discrete(heights):
+    return sum([(i > 1) for i in heights]) > .8 * len(heights)
+
+def count_var(x):
+    counts = {}
+    for val in x:
+        if val in counts:
+            counts[val] += 1
+        else:
+            counts[val] = 1
+    return counts
+    
+def compute_density(values):
+	density = gaussian_kde(values)
+	density.covariance_factor = lambda: 0.25
+	density._compute_covariance()
+	return density
+    
+def add_colorbar(fig, type, mappable, label):
+    if 'marginal' not in type:
+        caxes = fig.add_axes([0, 0.1, 0.05, 0.8])
+    else:
+        caxes = fig.add_axes([0, 0.1, 0.05, 0.57])
+    cbar = plt.colorbar(mappable=mappable, cax=caxes)
+    caxes.yaxis.set_ticks_position('left')
+    cbar.set_label(label)
+    caxes.yaxis.set_label_position('left')
+    return cbar, caxes
+
+def setup_tile(v, bins, discrete):
+    if not discrete:
+        v_lab = np.linspace(min(v), max(v) + 1, bins)
+        v_pos = range(len(v_lab))
+        v_vect = np.digitize(v, v_lab)
+    else:
+        v_lab = np.unique(v)
+        v_pos = range(len(v_lab))
+        v_map = dict(zip(v_lab, v_pos))
+        v_vect = np.vectorize(v_map.get)(v)
+    return v_vect, v_lab, v_pos
+
+def make_tile(x, y, bins, discrete_x, discrete_y, ax):
+    x_vect, x_lab, x_pos = setup_tile(x, bins, discrete_x)
+    y_vect, y_lab, y_pos = setup_tile(y, bins, discrete_y)
+    nums = len(x_vect)
+    counts = count_var(list(zip(y_vect, x_vect)))
+    intensity = np.zeros(shape=(len(y_lab), len(x_lab)))
         
+    for key, val in counts.items():
+        intensity[key] = val / nums
+    hm = ax.matshow(intensity, cmap='Blues', origin='lower', aspect='auto')
+    ax.xaxis.set_ticks_position('bottom')
+    if not discrete_x: x_lab = np.around(x_lab, decimals=1)
+    if not discrete_y: y_lab = np.around(y_lab, decimals=1)
+    setup_ticks(x_pos, x_lab, ax, 'x')
+    setup_ticks(y_pos, y_lab, ax, 'y')
+    return hm
+
+def setup_ticks(pos, lab, ax, axis):
```

Comment:
Please move `setup_ticks` either above or below all of the different `make_*` functions.